src/include/reconcile.i

```diff
@@ -166,21 +166,6 @@ __rec_page_time_stats(WT_SESSION_IMPL *session, WT_RECONCILE *r)
 static inline bool
 __wt_rec_need_split(WT_RECONCILE *r, size_t len)
 {
-    /*
-     * In the case of a row-store leaf page, trigger a split if a threshold number of saved updates
-     * is reached. This allows pages to split for update/restore and history store eviction when
-     * there is no visible data causing the disk image to grow.
-     *
-     * In the case of small pages or large keys, we might try to split when a page has no updates or
-     * entries, which isn't possible. To consider update/restore or history store information,
-     * require either page entries or updates that will be attached to the image. The limit is one
-     * of either, but it doesn't make sense to create pages or images with few entries or updates,
-     * even where page sizes are small (especially as updates that will eventually become overflow
-     * items can throw off our calculations). Bound the combination at something reasonable.
-     */
-    if (r->page->type == WT_PAGE_ROW_LEAF && r->entries + r->supd_next > 10)
-        len += r->supd_memsize;
-
```

Comment:
This is the main change.  We could throw away the rest of these changes and this would solve most of the problem.  The other changes are relatively small tweaks in comparison (but do have a positive impact on index size).